internal/integrations/integrations.go

```diff
@@ -0,0 +1,103 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package integrations // import "github.com/elastic/opentelemetry-collector-components/internal/integrations"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+var ErrNotFound = errors.New("not found")
+
+type Template interface {
+	Resolve(ctx context.Context, params map[string]any, pipelines []component.ID) (*Config, error)
+}
+
+type TemplateFinder interface {
+	FindTemplate(ctx context.Context, name, version string) (Template, error)
```

Comment:
Is the version parameter used in any of the extensions?

Reply:
Not used yet, we still have to think about versioning of templates.

I can remove it till we decide how to handle this.

Reply:
Sounds good, I would vote for removing it then. (a bit less code :))